From 6815aa4a2a0b2bd376d0f6e072580a9c60086e3e Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Thu, 14 Aug 2025 09:26:25 -0700
Subject: [PATCH] tests: Skip checking arch when testing sent body

The host architecture is included in the body of the request, skip
checking for the specific arch in order to keep the tests architecture
independent.

Related: RHEL-60137
---
 cmd/composer-cli/projects/info_test.go | 4 +++-
 cmd/composer-cli/projects/list_test.go | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/cmd/composer-cli/projects/info_test.go b/cmd/composer-cli/projects/info_test.go
index 3f9a85f..dd2a78c 100644
--- a/cmd/composer-cli/projects/info_test.go
+++ b/cmd/composer-cli/projects/info_test.go
@@ -457,7 +457,9 @@ func TestCmdProjectsInfoCloud(t *testing.T) {
 	sentBody, err := io.ReadAll(mcc.Req.Body)
 	assert.Nil(t, mcc.Req.Body.Close())
 	require.Nil(t, err)
-	assert.Contains(t, string(sentBody), `{"distribution":"homer","architecture":"x86_64","packages":["tmux"]}`)
+	// NOTE: sentBody also contains arch specific architecture, skip checking that.
+	assert.Contains(t, string(sentBody), `"distribution":"homer"`)
+	assert.Contains(t, string(sentBody), `"packages":["tmux"]`)
 	assert.Equal(t, "application/json", mcc.Req.Header.Get("Content-Type"))
 	assert.Equal(t, "/api/image-builder-composer/v2/search/packages", mcc.Req.URL.Path)
 }
diff --git a/cmd/composer-cli/projects/list_test.go b/cmd/composer-cli/projects/list_test.go
index b809224..386b793 100644
--- a/cmd/composer-cli/projects/list_test.go
+++ b/cmd/composer-cli/projects/list_test.go
@@ -363,7 +363,9 @@ func TestCmdProjectsListCloud(t *testing.T) {
 	sentBody, err := io.ReadAll(mcc.Req.Body)
 	assert.Nil(t, mcc.Req.Body.Close())
 	require.Nil(t, err)
-	assert.Contains(t, string(sentBody), `{"distribution":"homer","architecture":"x86_64","packages":["*"]}`)
+	// NOTE: sentBody also contains arch specific architecture, skip checking that.
+	assert.Contains(t, string(sentBody), `"distribution":"homer"`)
+	assert.Contains(t, string(sentBody), `"packages":["*"]`)
 	assert.Equal(t, "application/json", mcc.Req.Header.Get("Content-Type"))
 	assert.Equal(t, "/api/image-builder-composer/v2/search/packages", mcc.Req.URL.Path)
 }
-- 
2.50.1

